src/extract/path/mod.rs

```diff
@@ -0,0 +1,121 @@
+mod de;
+
+use super::{rejection::*, FromRequest};
+use crate::{extract::RequestParts, routing::UrlParams};
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+use std::ops::{Deref, DerefMut};
+
+/// Extractor that will get captures from the URL and parse them using [`serde`](https://crates.io/crates/serde).
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use axum::{extract::Path, prelude::*};
+/// use uuid::Uuid;
+///
+/// async fn users_teams_show(
+///     Path((user_id, team_id)): Path<(Uuid, Uuid)>,
+/// ) {
+///     // ...
+/// }
+///
+/// let app = route("/users/:user_id/team/:team_id", get(users_teams_show));
+/// # async {
+/// # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
+/// # };
+/// ```
+///
+/// If the path contains only one parameter, then you can omit the tuple.
+///
+/// ```rust,no_run
+/// use axum::{extract::Path, prelude::*};
+/// use uuid::Uuid;
+///
+/// async fn user_info(Path(user_id): Path<Uuid>) {
+///     // ...
+/// }
+///
+/// let app = route("/users/:user_id", get(user_info));
+/// # async {
+/// # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
+/// # };
+/// ```
+///
+/// Path segments also can be deserialized into any type that implements [serde::Deserialize](https://docs.rs/serde/1.0.127/serde/trait.Deserialize.html).
+/// Path segment labels will be matched with struct field names.
+///
+/// ```rust,no_run
+/// use axum::{extract::Path, prelude::*};
+/// use serde::Deserialize;
+/// use uuid::Uuid;
+///
+/// #[derive(Deserialize)]
+/// struct Request {
+///     user_id: Uuid,
+///     team_id: Uuid,
+/// }
+///
+/// async fn users_teams_show(
+///     Path(Request { user_id, team_id }): Path<(Uuid, Uuid)>,
```

Comment:
```suggestion
/// struct Params {
///     user_id: Uuid,
///     team_id: Uuid,
/// }
///
/// async fn users_teams_show(
///     Path(Params { user_id, team_id }): Path<Params>,
```

Request might be confusing since it conflicts with `http::Request`.